File: src/main/java/com/dersaun/apigestaocontas/domain/dtos/ResumoFormaPagamentoMensalResponseDTO.java
package com.dersaun.apigestaocontas.domain.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ResumoFormaPagamentoMensalResponseDTO implements Serializable {
    private ResumoFormaPagamentoMensalDTO cartao;
    private ResumoFormaPagamentoMensalDTO dinheiro;
    private ResumoFormaPagamentoMensalDTO parcelado;
    private ResumoFormaPagamentoMensalDTO total;

}

File: src/app/conta/dtos/resumo-forma-pagamento-mensal-response.dto.ts
import { ResumoFormaPagamentoMensalDTO } from './resumo-forma-pagamento-mensal.dto';

export class ResumoFormaPagamentoMensalResponseDTO {
  cartao: ResumoFormaPagamentoMensalDTO;
  dinheiro: ResumoFormaPagamentoMensalDTO;
  parcelado: ResumoFormaPagamentoMensalDTO;
  total: ResumoFormaPagamentoMensalDTO;

  constructor(dto?: Partial<ResumoFormaPagamentoMensalResponseDTO>) {
    if (dto) {
      Object.assign(this, dto);
    }
  }
}
